src/setup-elixir.js

```diff
@@ -103,71 +362,48 @@ function getVersionFromSpec(spec, versions) {
   }
 }
 
-async function getOtpVersions(osVersion) {
-  const result = await get(
-    `https://repo.hex.pm/builds/otp/${osVersion}/builds.txt`
-  )
-
-  return result
-    .trim()
-    .split('\n')
-    .map(line => {
-      const match = line.match(/^OTP-([\.\d]+)/)
+function getRunnerOSVersion() {
+  const mapToUbuntuVersion = {
+    ubuntu16: 'ubuntu-16.04',
+    ubuntu18: 'ubuntu-18.04',
+    ubuntu20: 'ubuntu-20.04',
+  }
 
-      if (match) {
-        const [_, version] = match
-        return version
-      } else {
-        return line
-      }
-    })
+  return mapToUbuntuVersion[process.env.ImageOS] || 'ubuntu-18.04'
 }
 
-async function getElixirVersions() {
-  const result = await get('https://repo.hex.pm/builds/elixir/builds.txt')
-  const map = new Map()
-
-  result
-    .trim()
-    .split('\n')
-    .forEach(line => {
-      const match =
-        line.match(/^(v\d+\.\d+\.\d+(?:-.+)?)-otp-(\d+)/) ||
-        line.match(/^([^-]+)-otp-(\d+)/)
+function get(urlOrFile) {
+  return new Promise(async (resolve, reject) => {
+    if (urlOrFile.remote) {
+      const req = https.get(urlOrFile.remote, {
+        headers: {'user-agent': 'setup-elixir'},
```

Comment:
Nice, this should help with stats.

Reply:
I didn't actually do it for that 😄, but because local tests failed. I'm glad it helps in that case, too.